### PR TITLE
AnnotateTraits: add deleteAttribute function

### LINKED ISSUE #213: Create a deleteAttribute in Annotate_Traits
Is there a way how to delete or rewrite an attribute using HighFive?

### PR DESCRIPTION
**Pull request by alkino**
Fix #213

### CODE REVIEW

File: tests/unit/tests_high_five_base.cpp [hunk 1]
```diff
@@ -651,25 +651,21 @@ void readWriteAttributeVectorTest() {
         Group g = file.createGroup("dummy_group");
 
         // check that no attributes are there
-        std::size_t n = g.getNumberAttributes();
-        BOOST_CHECK_EQUAL(n, 0);
+        BOOST_CHECK_EQUAL(g.getNumberAttributes(), 0);
```
Comment (ferdonline) [inline tests/unit/tests_high_five_base.cpp]: BOOST_CHECK_EQUAL, on error, this works pretty bad when trying to output a meaningful message.
I'd suggest to leave as it was (maybe changing the type to auto)

**Comment by ferdonline** [comment]
Maybe there was some problem merging, maybe a fast-forward, because we see some commits that shouldnt be here.

**Comment by alkino** [comment]
better like that @ferdonline ?

**Comment by ferdonline** [comment]
I am just wondering... maybe we should enforce good behavior, by checking (per the docs)
"This function should not be used when attribute identifiers are open on loc_id"

**Comment by alkino** [comment]
I cannot see a function to know if a attribute identifiers  is open.

**Comment by ferdonline** [comment]
https://support.hdfgroup.org/HDF5/doc1.8/RM/RM_H5I.html#Identify-GetRef

**Comment by alkino** [comment]
I think of it, but we don't know if internally they use some refs.

**Comment by alkino** [comment]
Can we merge it like this? @ferdonline

**Pull request merged**